Sources/SIL/SIL.swift

```diff
@@ -355,7 +368,7 @@ public enum TupleElements {
 }
 
 // Reverse-engineered from -emit-sil
-public indirect enum Type {
+public indirect enum Type: Equatable {
     case addressType(_ type: Type)
```

Comment:
* Enum case names don't need to repeat the enum type information. I'd suggest dropping the "Type" suffixes.
* No need for argument names.

`address(Type)`